sklearn/metrics/tests/test_pairwise.py

```diff
@@ -438,7 +439,44 @@ def test_cosine_distances():
     assert_true(np.all(D <= 2.))
 
 
-# Paired distances
+def test_gower_distances():
+    # Check the pairwise Gower distances computation
+
+    # Zero distance expected
+    X = np.array([[1, 4141.22, False, 'ABC'],
+                  [1, 4141.22, False, 'ABC']], dtype=object)
+
+    D = gower_distances(X)
+
+    assert_array_almost_equal(D, np.array([[0.0, 0.0],
+                                           [0.0, 0.0]]))
+
+    X = np.array([['M', False, 222.22, 1],
+                  ['F', True, 333.22, 2],
+                  ['M', True, 1934.0, 4],
+                  [None, None, np.nan, np.nan]], dtype=object)
+
+    D = gower_distances(X)
+
+    assert_array_almost_equal(D, np.array([[0.0, 0.599545, 0.75, np.nan],
+                                           [0.599545, 0.0, 0.650455, np.nan],
+                                           [0.75, 0.650455, 0.0, np.nan],
+                                           [np.nan, np.nan, np.nan, 0]]))
+
+    # Test to obtain a non-squared distance matrix
+    X = np.array([['Syria', 1200, 0.0, 411114.44, True],
+                  ['Ireland', 300, 0.0, 199393333.22, False],
+                  ['United Kingdom', 100, 0.0, 32323222.121, False]],
+                 dtype=object)
+
+    Y = np.array([['United Kingdom', 200, 0.0, 99923921.47, True]],
+                 dtype=object)
+
+    D = gower_distances(X, Y)
+
+    assert_array_almost_equal(D, np.array([[0.48184],
+                                           [0.51816],
+                                           [0.286128]]))
 
```

Comment:
I don't see any tests with X and Y being sparse in actuality.

Reply:
No support to sparse matrices for a while.